trivy_report/report_generator.py

```diff
@@ -6,15 +8,20 @@
 
 
 class VulnerabilityDict(TypedDict):
-    VulnerabilityID: str
-    PkgName: str
-    InstalledVersion: str
-    FixedVersion: str
+    RuleID: Optional[str]
```

Comment:
Why are all of these properties set as optional?
The code below does not actually handle them as optional.

If this is only done for the Secret handling then it is better to have a separate TypedDict for the secret entries.

Reply:
Creating another TypedDict would be a better solution.